### DISCUSSION TITLE
Always return response with status 200 when options.throwHttpErrors = false

### DISCUSSION THREAD

**Discussion by ansaru** [sindresorhus/ky#393] [Q&A] [answered]
Hi,
I use version 0.28.6.
When set **options.throwHttpErrors=false** in options and then get response after call:
 `resp = await ky(url, options);`
the resp always has status of 200.

Probably, it's a bug.

**Comment by sholladay** [comment]
I cannot reproduce this.

```js
const response = await ky.get('https://httpbin.org/status/500', { throwHttpErrors : false });
console.log(response.status);  // 500
```

You will need to submit a full bug report if you believe this is a problem with Ky (and if so, please do so on the issue tracker).

That said, I think we should update [this test](https://github.com/sindresorhus/ky/blob/2049affe2ca939f53350fb9b7354c3ff8d2e1c1f/test/main.ts#L375-L384) to explicitly check that a response is returned with the appropriate status.

**Comment by ansaru** [comment]
Thanks for the answer.

Here is the code that reproduces the error: 

```
const abortController = new AbortController();
    const abort = () => {
      abortController.abort();
    }

    const onDownloadProgressWatcher = (progressObj) => {
      console.log(`propgress: ${progressObj}`);
    }

    async function getError500() {
      const url = 'https://httpbin.org/status/500';
      const options = {
        signal : abortController.signal,
        onDownloadProgress : onDownloadProgressWatcher,
        timeout : 7000,
        method : 'get',
        throwHttpErrors : false,
        headers : { 'Accept': 'application/json' },
      };
      const response = await ky(url, options);

      console.log(response);
    }

```

**Reply by sholladay** [reply]
Looks like it's because of the use of `onDownloadProgress`. 

```js
const url = 'https://httpbin.org/status/500';
const options = {
    onDownloadProgress : () => {},
    throwHttpErrors : false
};
const response = await ky.get(url, options);
console.log(response.status);  // 200
```

Definitely a bug. I've opened #394 to track this.

**Comment by sholladay** [comment, marked as answer]
See #394 for any further updates.
